docs/rules/no-invalid-this.md

```diff
@@ -0,0 +1,221 @@
+# Disallow `this` keyword on the outside of constructors/methods (no-invalid-this)
+
+To access `this` keyword on the outside of constructors/methods is dangerous.
```

Comment:
I think this needs to be re-written. I'm not sure I'm getting the full picture of what this rule is supposed to flag. As far as I understand the point of the rule to flag usage of `this` keyword outside of classes (es6) or extendable objects (es3/es5).